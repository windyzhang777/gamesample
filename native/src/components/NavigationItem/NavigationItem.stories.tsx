import React from 'react';
import {storiesOf} from '@storybook/react-native';
import NavigationItem from '.';
import resolvePlatform from 'shared/utils/resolvePlatform';
import {NativeRouter} from 'react-router-native';

storiesOf('Navigation Item', module)
  .addDecorator((story: () => React.ReactNode) => <NativeRouter>{story()}</NativeRouter>)
  .add('Normal Navigation Item', () => {
    return (
      <NavigationItem
        toggleDrawerCallback={() => {
          console.log('Toggle Drawer Callback');
        }}
        navigationItem={{
          menuText: 'Notifications',
          key: 'notifications',
          linkTo: '/notifications',
          linkComponent: resolvePlatform('EmptyScreen', {
            emptyScreenName: 'Notifications',
          }),
        }}
      />
    );
  })
  .add('Navigation Item No Underline', () => {
    return (
      <NavigationItem
        toggleDrawerCallback={() => {
          console.log('Toggle Drawer Callback');
        }}
        navigationItem={{
          menuText: 'Notifications',
          key: 'notifications',
          linkTo: '/notifications',
          linkComponent: resolvePlatform('EmptyScreen', {
            emptyScreenName: 'Notifications',
          }),
          removeBottomUnderline: true,
        }}
      />
    );
  });
