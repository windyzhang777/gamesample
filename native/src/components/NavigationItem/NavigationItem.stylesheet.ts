import {StyleSheet} from 'react-native';
import {navigationPrimaryColor} from 'shared/theme/variables';

export default StyleSheet.create({
  NavigationItemText: {
    backgroundColor: navigationPrimaryColor,
    alignContent: 'center',
    fontSize: 18,
    textTransform: 'uppercase',
    fontWeight: 'bold',
    color: 'white',
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 10,
  },
  NavigationItemTextUnderline: {
    backgroundColor: '#4A5696',
    height: 1,
  },
});
