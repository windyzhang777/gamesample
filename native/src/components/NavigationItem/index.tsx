import React from 'react';
import NavigationItemShared, {NavigationItemProps} from 'shared/containers/NavigationItem';
import {Text, View} from 'react-native';
import styles from './NavigationItem.stylesheet';
import {Link} from 'react-router-native';

/**
 * This is a native version of rendering
 * @param {NavigationItemProps} props
 */
export default function NavigationItem(props: NavigationItemProps) {
  //This is just passing down the properties
  return (
    <NavigationItemShared
      {...props}
      renderComponent={({navigationItem, toggleDrawerCallback}: NavigationItemProps) => {
        // This is where the native rendering is happening
        return (
          <>
            <Link
              to={
                navigationItem.linkTo
                  ? navigationItem.linkTo
                  : (() => {
                      console.warn(
                        'NavigationItem(' +
                          navigationItem.menuText +
                          "): no linkTo provided, default to '/'",
                      );
                      return '/';
                    })()
              }
              onPress={() => {
                toggleDrawerCallback !== undefined
                  ? toggleDrawerCallback()
                  : (() => {
                      console.log('Link Pressed but no toggle callback');
                    })();
              }}>
              <Text style={styles.NavigationItemText}>
                {navigationItem.icon ? navigationItem.icon : ''} {navigationItem.menuText}
              </Text>
            </Link>
            {navigationItem.removeBottomUnderline ? (
              <></>
            ) : (
              <View style={styles.NavigationItemTextUnderline} />
            )}
          </>
        );
      }}
    />
  );
}
