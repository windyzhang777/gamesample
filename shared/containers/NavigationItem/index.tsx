import {RenderedComponent} from '../../models/RenderedComponent';
import {NavigationItemType} from '../../store/navigation-system/types';

/**
 * These are the properties of the NavigationItemShared
 * These are passed to the dummy rendering component
 */
export interface NavigationItemProps {
  navigationItem: NavigationItemType;
  toggleDrawerCallback: () => void;
}

/**
 * This is the navigation item's logic
 * @param {NavigationItemProps} props properties for shared
 */
export default function NavigationItem({
  navigationItem,
  toggleDrawerCallback,
  renderComponent,
}: NavigationItemProps & RenderedComponent<NavigationItemProps>) {
  /**
   * This is where we would implement our react logic
   * Such as using hooks
   */

  /**
   * This contains all the render properties
   * for what data we pass to be rendered
   */
  const renderProps: NavigationItemProps = {
    navigationItem,
    toggleDrawerCallback,
  };
  return renderComponent(renderProps);
}
