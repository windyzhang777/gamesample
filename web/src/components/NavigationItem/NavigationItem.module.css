.NavigationItem {
  text-align: start;
  height: 50px;
  cursor: pointer;
  list-style: none;
}

.NavigationItem a {
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;
}
