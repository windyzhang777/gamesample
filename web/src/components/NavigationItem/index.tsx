import React from 'react';
import NavigationItemShared, {NavigationItemProps} from 'shared/containers/NavigationItem';
import classes from './NavigationItem.module.css';
import {Link} from 'react-router-dom';

/**
 * This is a web version of rendering
 * @param {NavigationItemProps} props
 */
export default function NavigationItem(props: NavigationItemProps) {
  //This is just passing down the properties
  return (
    <NavigationItemShared
      {...props}
      renderComponent={({navigationItem, toggleDrawerCallback}: NavigationItemProps) => {
        // This is where the web rendering is happening
        // needUpdate: <Link to={link} onClick={closeDrawerCallback}> closeDrawerCallback should be local

        return (
          <li className={classes.NavigationItem}>
            <Link to={navigationItem.linkTo} onClick={toggleDrawerCallback}>
              {navigationItem.menuText}
            </Link>
          </li>
        );
      }}
    />
  );
}
