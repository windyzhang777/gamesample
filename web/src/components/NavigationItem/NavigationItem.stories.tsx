import React from 'react';
import {storiesOf} from '@storybook/react';
import NavigationItem from '.';
import resolvePlatform from 'shared/utils/resolvePlatform';
import {BrowserRouter} from 'react-router-dom';

storiesOf('Navigation Item', module)
  .addDecorator((story) => <BrowserRouter>{story()}</BrowserRouter>)
  .add('Normal Navigation Item', () => (
    <div style={{backgroundColor: '#303d88', height: '100vh'}}>
      <NavigationItem
        navigationItem={{
          menuText: 'Notifications',
          key: 'notifications',
          linkTo: '/notifications',
          linkComponent: resolvePlatform('EmptyScreen', {
            emptyScreenName: 'Notifications',
          }),
        }}
        toggleDrawerCallback={() => {
          console.log('toggleDrawerCallback');
        }}
      />
    </div>
  ));
