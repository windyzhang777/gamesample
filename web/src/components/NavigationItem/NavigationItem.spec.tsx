import React from 'react';
import {shallow} from 'enzyme';
import NavigationItem from './index';

const render = (props: any) => shallow(<NavigationItem {...props} />);

describe('NavigationItem Component', () => {
  it('renders with default props', () => {
    expect(render({})).toMatchSnapshot();
  });
});
