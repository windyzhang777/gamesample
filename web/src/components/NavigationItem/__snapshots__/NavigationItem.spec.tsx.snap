// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem Component renders with default props 1`] = `
<NavigationItem
  renderComponent={[Function]}
/>
`;
